core/tee/tee_obj.c

```diff
@@ -53,6 +53,25 @@ TEE_Result tee_obj_get(struct tee_ta_ctx *ctx, uint32_t obj_id,
 	return TEE_ERROR_BAD_PARAMETERS;
 }
 
+TEE_Result tee_obj_get1(struct tee_ta_ctx *ctx, uint32_t obj_id,
```

Comment:
This function is only used as a helper function by 3 new xxx1 function. tee_obj_get() still remains and will still remain even after the deprecated functions replaced by the 3 new xxx1 functions are removed.

The old tee_obj_get() is a function without side effects (only use to lookup the pointer to a struct tee_obj based on a handle from the TA) while this has side effects. I would prefer if we didn't add this function and instead checked the objectType inside the 3 new xxx1 functions.